src/Unread.ts

```diff
@@ -55,6 +55,11 @@ export function eventTriggersUnreadCount(ev: MatrixEvent): boolean {
 }
 
 export function doesRoomHaveUnreadMessages(room: Room): boolean {
+    // if there is no timeline at all, e.g. by message retention
+    if (room.timeline.length === 0) {
+        return false;
+    }
```

Comment:
This is insufficient, just because the client's loaded timeline has length 0, doesn't mean it has yet tried to backpaginate to see if there are any events. The server sends an unread count for a reason